### PR TITLE
#195 - Migrate to swagger 3.0 once swagger-codegen has stable release

### PR DESCRIPTION
**Pull request by duttarnab**
#195 - Migrate to swagger 3.0 once swagger-codegen has stable release
https://github.com/GluuFederation/oxd/issues/195

### CODE REVIEW

File: oxd-gen-client/pom.xml [hunk 1]
```diff
@@ -1,292 +1,291 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-    <modelVersion>4.0.0</modelVersion>
-    <artifactId>oxd-gen-client</artifactId>
-    <packaging>jar</packaging>
-    <name>oxd-gen-client</name>
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (yuriyz) [inline oxd-gen-client/pom.xml]: We maintain pom.xml. Here it is not formatter and it seems some plugin configurations are changed. Do all tests pass with this PR ?

File: oxd-gen-client/src/main/java/io/swagger/client/Pair.java [hunk 2]
```diff
@@ -2,19 +2,17 @@
...
  * https://github.com/swagger-api/swagger-codegen.git
  * Do not edit the class manually.
  */
 
-
 package io.swagger.client;
 
-@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2019-07-23T08:29:06.952Z")
-public class Pair {
+@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2020-02-13T06:19:07.893Z[GMT]")public class Pair {
```
Comment (yuriyz) [inline oxd-gen-client/src/main/java/io/swagger/client/Pair.java]: Would you please check whether there is way to skip `date` during generation? Hopefully generator has such option. These dates changes creates a lot of noise.

File: oxd-gen-client/src/main/java/io/swagger/client/ProgressResponseBody.java [hunk 3]
```diff
@@ -2,15 +2,14 @@
  * oxd-server
  * oxd-server
  *
- * OpenAPI spec version: 4.0.beta
+ * OpenAPI spec version: 4.0
```
Comment (yuriyz) [inline oxd-gen-client/src/main/java/io/swagger/client/ProgressResponseBody.java]: We should have 4.2 here.

File: oxd-server/src/main/resources/swagger.yaml [hunk 4]
```diff
@@ -1,1961 +1,2482 @@
...
     name: License
     url: https://github.com/GluuFederation/oxd/blob/master/LICENSE
-
-host: gluu.org
-basePath: /
-schemes:
- - https
-
-# tags are used for organizing operations
+  version: "4.0"
```
Comment (yuriyz) [inline oxd-server/src/main/resources/swagger.yaml]: 4.2

File: oxd-gen-client/README.md [hunk 5]
```diff
@@ -1,35 +1,47 @@
 # swagger-java-client
 
+oxd-server
+- API version: 4.0
```
[the remark this answers is not in the document]
Comment (duttarnab) [reply oxd-gen-client/README.md]: Done

File: oxd-gen-client/pom.xml (hunk 1 above)
Comment (duttarnab) [reply oxd-gen-client/pom.xml]: We have formatted pom.xml. All test cases are running.

File: oxd-gen-client/src/main/java/io/swagger/client/Pair.java (hunk 2 above)
Comment (duttarnab) [reply oxd-gen-client/src/main/java/io/swagger/client/Pair.java]: This is only possible by generating  Java Client Offline with `-D hideGenerationTimestamp=true`.
We have done that by following below steps. Now there is no `date` in generated code.

1. Dowloaded codegen stable version source code from https://github.com/swagger-api/swagger-codegen/releases/tag/v3.0.16

2. Build it

3. Generate java client using below command
`java -jar swagger-codegen-cli.jar generate -i swagger.yaml -D hideGenerationTimestamp=true -l java`

File: oxd-gen-client/src/main/java/io/swagger/client/ProgressResponseBody.java (hunk 3 above)
Comment (duttarnab) [reply oxd-gen-client/src/main/java/io/swagger/client/ProgressResponseBody.java]: done

File: oxd-server/src/main/resources/swagger.yaml (hunk 4 above)
Comment (duttarnab) [reply oxd-server/src/main/resources/swagger.yaml]: done

**Pull request merged**